.github/workflows/ci.yml

```diff
@@ -116,6 +127,35 @@ jobs:
           cd ${GITHUB_WORKSPACE}/build && pwd
           du -hcs _deps/
 
+      - name: Configure CMake for sanitizers build
+        if: matrix.sanitizers == 'Sanitizers'
+        run: |
+          apt -y update
+          apt -y upgrade
+          apt install -y clang
+          which clang
+          echo "ulimit is"
+          ulimit -s
+          echo "-----------------------------"
+          echo "disk space is:"
+          df -h
+          echo "-----------------------------"
+          mkdir -p $GITHUB_WORKSPACE/build
+          cd $GITHUB_WORKSPACE/build
+          cmake .. \
```

Comment:
I believe we can make only one build step for sanitizers and non sanitizers and control the configure all the flags in under the matrix

Reply:
Another downside of this: `[build (alpine-dev:latest, Release, g++, gcc, -Werror, NoSanitizers, OFF, OFF)](https://github.com/dragonflydb/dragonfly/actions/runs/12804933055/job/35700274090?pr=4462#logs)` 

See https://github.com/dragonflydb/dragonfly/actions/runs/12804933055/job/35700274090?pr=4462 

(I will figure out something :smile: -- need to expiriment )

Reply:
ok I suggest use https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow to understand what is the correct why to do this